File: solution/0000-0099/0003.Longest Substring Without Repeating Characters/Solution.ts
function lengthOfLongestSubstring(s: string): number {
    const cnt = new Map<string, number>();
    let ans = 0;
    for (let l=0, r=0; r < s.length; ++r) {
        cnt.set(s[r], (cnt.get(s[r]) || 0) + 1);
        while (cnt.get(s[r])! > 1) {
            cnt.set(s[l], cnt.get(s[l])! - 1);
            ++l;
        }
        ans = Math.max(ans, r-l+1);
    }
    return ans;
};

File: solution/0000-0099/0003.Longest Substring Without Repeating Characters/Solution.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int[] cnt = new int[128];
        int ans = 0;
        for (int l=0, r=0; r < s.length(); ++r) {
            char c = s.charAt(r);
            ++cnt[c];
            while (cnt[c] > 1) {
                --cnt[s.charAt(l++)];
            }
            ans = Math.max(ans, r-l+1);
        }
        return ans;
    }
}
